app/src/main/kotlin/com/exallium/todoapp/MainActivity.kt

```diff
@@ -59,4 +66,9 @@ class MainActivity : AppCompatActivity() {
             super.onBackPressed()
         }
     }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        onBackPressed()
```

Comment:
we should be making sure that what we hit is actually the up button, and in this case handling it appropriately and returning true.